src/ladds/particle/Constellation.cpp

```diff
@@ -69,6 +69,11 @@ Constellation::Constellation(ConfigReader &constellationConfig, ConfigReader &co
       schedule[i].push_back(timestamps[i] + j * timeStepSize);
     }
   }
+
+  // if checkpoint is loaded, throw away already inserted satellites
+  if (config.defines("io/hdf5/checkpoint/file")) {
+    tick(config.get<size_t>("io/hdf5/checkpoint/iteration", -1, true));
+  }
```

Comment:
I needed the `startingIteration` as it is in the simulation class, but instead of passing it, I copied what was assigned to `startingIteration.` I see now though that this is false, because the fallback value of -1 used in Simulation would be added +1 and would then be 0 (right?). I mean this snippet:

`  const auto startingIteration =
      config.defines("io/hdf5", true) ? config.get<size_t>("io/hdf5/checkpoint/iteration", -1, true) + 1 : 0;`

Is there actually a case where the -1 (+1) becomes relevant? Because if a checkpoint is used, but no iteration is given, the highest iteration found in the checkpoint is always loaded right?

Reply:
I think I casted `simulationTime` to long, because startTime can be negative (if insertion started before referenceTime or before the startingIteration when a checkpoint is loaded) and the negative number was then evaluated to be bigger than `simulationTime` since both numbers were apparently treated as `size_t` numbers. So the `if` in Line 111 when loading a checkpoint is true if the loaded iteration is later than the scheduled start of the constellation (insertion started during simulation that produced the loaded checkpoint)